liblangutil/SemVerHandler.cpp

```diff
@@ -105,8 +105,19 @@ bool SemVerMatchExpression::MatchComponent::matches(SemVerVersion const& _versio
 				didCompare = true;
 				cmp = _version.numbers[i] - version.numbers[i];
 			}
+
 		if (cmp == 0 && !_version.prerelease.empty() && didCompare)
```

Comment:
I think this is not the right place to add this, because this `if` implements the logic that prerelease versions are always strictly smaller than the version without prerelease.

Reply:
But 0.5.3-pre is not strictly less than 0.5. So when the levelsPresent is less than 3, the cmp would be 0. And after checking the prerelease, it would be decremented to -1, which is not expected. This added lines is trying to fix that.

If you still don't like it being here, I'd have to change the use of levelsPresent.